runtime/interpreter/interpreter_expression.go

```diff
@@ -523,6 +497,83 @@ func (interpreter *Interpreter) testEqual(left, right Value, expression *ast.Bin
 	)
 }
 
+func (interpreter *Interpreter) testComparison(left, right Value, expression *ast.BinaryExpression) BoolValue {
+	left = interpreter.Unbox(
+		LocationRange{
+			Location:    interpreter.Location,
+			HasPosition: expression.Left,
+		},
+		left,
+	)
+
+	right = interpreter.Unbox(
+		LocationRange{
+			Location:    interpreter.Location,
+			HasPosition: expression.Right,
+		},
+		right,
+	)
+
+	leftComparable, ok := left.(ComparableValue)
+	if !ok {
+		return FalseValue
+	}
+
+	rightComparable, ok := right.(ComparableValue)
+	if !ok {
+		return FalseValue
+	}
+
+	switch expression.Operation {
+	case ast.OperationLess:
+		return leftComparable.Less(
+			interpreter,
+			rightComparable,
+			LocationRange{
+				Location:    interpreter.Location,
+				HasPosition: expression,
+			},
```

Comment:
Maybe refactor the common location range construction here and in the other cases into a variable declaration before the switch

Reply:
Done.